ragengine/tests/vector_store/test_faiss_store.py

```diff
@@ -0,0 +1,225 @@
+    # Test that refreshing existing documents prints error.
+    original_documents = [Document(doc_id="1", text="First document", metadata={"type": "text"})]
+    vector_store_manager.index_documents("test_index", original_documents)
+
+    new_documents = [Document(doc_id="1", text="Updated document", metadata={"type": "text"}),
+                     Document(doc_id="2", text="Second document", metadata={"type": "text"})]
+
+    refresh_results = vector_store_manager.refresh_documents("test_index", new_documents)
+
+    captured = capsys.readouterr()
+
+    # Check if the NotImplementedError message was printed
+    assert "Refresh not yet fully implemented for index" in captured.out
+    assert not refresh_results
+"""
+
+def test_persist_and_load_index_store(vector_store_manager):
+    """Test that the index store is persisted and loaded correctly."""
+    # Add a document and persist the index
+    documents = [Document(doc_id="1", text="Test document", metadata={"type": "text"})]
+    vector_store_manager.index_documents("test_index", documents)
+    vector_store_manager._persist("test_index")
+
+    # Simulate a fresh load of the index store (clearing in-memory state)
+    vector_store_manager.index_store = None  # Clear current in-memory store
+    vector_store_manager._load_index_store()
+
+    # Verify that the store was reloaded and contains the expected index structure
+    assert vector_store_manager.index_store is not None
+    assert len(vector_store_manager.index_store.index_structs()) > 0
+
+# TODO: Prevent default re-indexing from load_index_from_storage
+def test_persist_and_load_index(vector_store_manager):
+    """Test that an index is persisted and then loaded correctly."""
+    # Add a document and persist the index
+    documents = [Document(doc_id="1", text="Test document", metadata={"type": "text"})]
+    vector_store_manager.index_documents("test_index", documents)
+
+    documents = [Document(doc_id="1", text="Another Test document", metadata={"type": "text"})]
+    vector_store_manager.index_documents("another_test_index", documents)
+
+    vector_store_manager._persist_all()
+
+    # Simulate a fresh load of the index (clearing in-memory state)
+    vector_store_manager.index_map = {}  # Clear current in-memory index map
+    loaded_indices = vector_store_manager._load_indices()
+
+    # Verify that the index was reloaded and contains the expected document
+    assert loaded_indices is not None
+    assert vector_store_manager.document_exists("test_index", "1")
+    assert vector_store_manager.document_exists("another_test_index", "1")
+
+    vector_store_manager.index_map = {}  # Clear current in-memory index map
+    loaded_index = vector_store_manager._load_index("test_index")
+
+    assert loaded_index is not None
+    assert vector_store_manager.document_exists("test_index", "1")
+    assert not vector_store_manager.document_exists("another_test_index", "1") # Since we didn't load this index
+
```

Comment:
how to run this test

Reply:
`pytest -o log_cli=true -o log_cli_level=INFO ragengine/tests`
next PR updates test.yml (unit-tests) and makefile to include tests